libs/editor/egui_editor/src/theme.rs

```diff
@@ -0,0 +1,139 @@
+pub const GRAY: ThemedColor =
+    ThemedColor { light: Color32::from_rgb(142, 142, 147), dark: Color32::from_rgb(142, 142, 147) };
+pub const GRAY_2: ThemedColor =
+    ThemedColor { light: Color32::from_rgb(174, 174, 178), dark: Color32::from_rgb(99, 99, 102) };
+pub const GRAY_3: ThemedColor =
+    ThemedColor { light: Color32::from_rgb(199, 199, 204), dark: Color32::from_rgb(72, 72, 74) };
+pub const GRAY_4: ThemedColor =
+    ThemedColor { light: Color32::from_rgb(209, 209, 214), dark: Color32::from_rgb(58, 58, 60) };
+pub const GRAY_5: ThemedColor =
+    ThemedColor { light: Color32::from_rgb(229, 229, 234), dark: Color32::from_rgb(44, 44, 46) };
+pub const GRAY_6: ThemedColor =
+    ThemedColor { light: Color32::from_rgb(242, 242, 247), dark: Color32::from_rgb(28, 28, 30) };
+pub const WHITE: ThemedColor = ThemedColor { light: Color32::WHITE, dark: Color32::BLACK };
+
+/// provides a mechanism for the application developer to override colors for dark mode and light
+/// mode and for us to provide defaults
+#[derive(Default)]
+pub struct VisualAppearance {
+    pub current_theme: Theme,
+
+    pub text: Option<ThemedColor>,
+    pub selection_bg: Option<ThemedColor>,
+    pub heading: Option<ThemedColor>,
+    pub heading_line: Option<ThemedColor>,
+    pub code: Option<ThemedColor>,
+    pub bold: Option<ThemedColor>,
+    pub italics: Option<ThemedColor>,
+    pub strikethrough: Option<ThemedColor>,
+    pub link: Option<ThemedColor>,
+}
+
+impl VisualAppearance {
+    pub fn text(&self) -> Color32 {
+        self.text.unwrap_or(BLACK).get(self.current_theme)
+    }
+
+    pub fn selection_bg(&self) -> Color32 {
+        let mut color = BLUE;
+
+        // light mode: use half saturation
+        color.light = {
+            let mut color_hsva = Hsva::from(color.light);
+            color_hsva.s /= 2.0;
+            Color32::from(color_hsva)
+        };
+
+        self.selection_bg.unwrap_or(color).get(self.current_theme)
+    }
+
+    pub fn heading(&self) -> Color32 {
+        self.heading.unwrap_or(BLACK).get(self.current_theme)
+    }
+
+    pub fn heading_line(&self) -> Color32 {
+        self.heading_line.unwrap_or(GRAY).get(self.current_theme)
+    }
+
+    pub fn code(&self) -> Color32 {
+        self.code.unwrap_or(PINK).get(self.current_theme)
+    }
+
+    pub fn bold(&self) -> Color32 {
+        self.bold.unwrap_or(BLACK).get(self.current_theme)
+    }
+
+    pub fn italics(&self) -> Color32 {
+        self.italics.unwrap_or(BLACK).get(self.current_theme)
+    }
+
+    pub fn strikethrough(&self) -> Color32 {
+        self.strikethrough.unwrap_or(PINK).get(self.current_theme)
+    }
+
+    pub fn link(&self) -> Color32 {
+        self.link.unwrap_or(BLUE).get(self.current_theme)
+    }
+
+    pub fn update(&mut self, ui: &egui::Ui) {
+        if ui.visuals().dark_mode {
+            self.current_theme = Theme::Dark
+        } else {
+            self.current_theme = Theme::Light
+        }
```

Comment:
`self.current_theme = if { .. } else { .. };` would be a bit more concise

Reply:
bb24b906